### PR TITLE
double check the signing tests; set status so correct ReviewHelper is used

### LINKED ISSUE #3314: New unlisted submissions end up with wrong add-on status
New unlisted files are automatically approved, but the add-on status is still `STATUS_NOMINATED`instead of `STATUS_PUBLIC`.

### PR DESCRIPTION
**Pull request by eviljeff**
I'm taking a guess this will fix #3314, as I'm flying a little blind.  If it doesn't I'll revert

### CODE REVIEW

File: src/olympia/devhub/tasks.py [hunk 1]
```diff
@@ -110,9 +110,9 @@ def create_version_for_upload(addon, upload,
         # review since listed and sideload aren't supported for creation yet.
         if addon.status == amo.STATUS_NULL:
             if disallow_preliminary_review:
-                addon.update(status=amo.STATUS_PUBLIC)
+                addon.update(status=amo.STATUS_NOMINATED)
```
Comment (diox) [inline src/olympia/devhub/tasks.py]: So, hold on. Does that mean that, without this patch, we set add-ons as public without a review somehow ?

In addition, if that fixes your bug, then why was the issue only happening in the API ?

File: src/olympia/devhub/tasks.py [hunk 2]
```diff
@@ -110,9 +110,9 @@ def create_version_for_upload(addon, upload,
...
             else:
-                addon.update(status=amo.STATUS_LITE)
+                addon.update(status=amo.STATUS_PENDING)
```
Comment (diox) [inline src/olympia/devhub/tasks.py]: Isn't `STATUS_PENDING` for personas ? Any anway, wouldn't you want `STATUS_LITE` in that case, like it was before ? Wait, scratch that, why isn't it `STATUS_UNREVIEWED` ?

File: src/olympia/devhub/tasks.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/devhub/tasks.py]: > So, hold on. Does that mean that, without this patch, we set add-ons as public without a review somehow ?
> In addition, if that fixes your bug, then why was the issue only happening in the API ?

We do indeed.  But, devhub uploads don't follow this path - the parameter `submit` in `views/handle_upload` is only set to True during an API call, so the sequence of calls that create the version and set the status here don't happen, and the Addon status is set in `views/submit_addon` (for unlisted) or `views/submit_license` (listed) instead.  (Yes, it's totally non-trivial to discover this; yes, it's crazy that we have upload code that is not used in the actual devhub in `devhub/views.py`)

Comment (eviljeff) [inline src/olympia/devhub/tasks.py]: I'm fairly certain we'd have discovered this bug earlier if the signing API supported sideloaded add-ons btw.

File: src/olympia/devhub/tasks.py (hunk 2 above)
Comment (eviljeff) [inline src/olympia/devhub/tasks.py]: You're correct, it should be STATUS_UNREVIEWED.  I've not worked out why STATUS_LITE is somehow okay, but STATUS_PUBLIC is not.  Might be safer to leave it as STATUS_LITE if it's always worked this way?

File: src/olympia/devhub/tasks.py (hunk 1 above)
Comment (diox) [inline src/olympia/devhub/tasks.py]: Woah, ok. I followed the thread to make sure the same method was used, but I missed that `submit` parameter. Sigh.

File: src/olympia/devhub/tasks.py (hunk 2 above)
Comment (diox) [inline src/olympia/devhub/tasks.py]: Yeah, I'd leave `STATUS_LITE` since this is going away soon anyway.

**Comment by diox** [comment]
r+

**Pull request merged**